src/routes/auth/signin-user-by-email.e2e.test.js

```diff
@@ -0,0 +1,139 @@
+		}
+	})
+	describe('Failure', async() => {
+		it(`should return a 400 error if request body is absent`, async() => {
+			const expectedError = {
+				"error": {
+					"message": 
+				"email is required, password is required"
+				},
+				"status": "failed"
+			}
+
+			return fixtures.api()
+				.post('/api/v1/auth/signin')
+				.expect(400, expectedError)
+		})
+		it(`should return a 400 error if email is absent`, async() => {
+			const expectedError = {
+				"error": {
+					"message": "email is required"
+				},
+				"status": "failed"
+			}
+
+			return fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					password : signinInfo.password
+				})
+				.expect(400, expectedError)
+		})
+		it(`should return a 400 error password is absent`, async() => {
+			const expectedError = {
+				"error": {
+					"message": "password is required"
+				},
+				"status": "failed"
+			}
+
+			return fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					email: signinInfo.email
+				})
+				.expect(400, expectedError)
+		})
+		it('should return a 400 error if email is invalid', async () => {
+			const expectedError = {
+				"error": {
+					"message": "email must be a valid email"
+				},
+				"status": "failed"
+			}
+			return fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					email: 'invalid-email.com',
+					password: signinInfo.password
+				})
+				.expect(400, expectedError)
+		})
+
+		it('should return an error if password is invalid', async () => 
+			fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					email: signinInfo.email,
+					password: "invalid-password"
+				})
+				.expect(401)
+				.then((res) => {
+					expect(res.body.error.message)
+						.to.eql(`Invalid email or password.`)
+					expect(res.body.error.status).to.eql("fail")
+				})
+		)
+	})
+
+	describe('Success', () => {
+		it('should return 201', async () => 
+			fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					email: signinInfo.email,
+					password: signinInfo.password
+				})
+				.expect('Content-Type', /json/)
+				.expect(200)
+		)
+		it('should return the right response', async () => {			
+			fixtures.api()
+				.post('/api/v1/auth/signin')
+				.send({
+					email: signinInfo.email,
+					password: signinInfo.password
+				})
+				.expect(200)
+				.then((res) => {
+					expect(res.body.data.accessToken).to.be.a('string')
+					expect(res.body.data.refreshToken).to.be.an('string')
+					delete res.body.data.accessToken
+					delete res.body.data.refreshToken
+					expect(res.body).to.eql({
+						status: 'success',
+						data: {
+							userId: user.id,
+							firstName: user.firstName,
+							lastName: user.lastName,
+							email: user.email,
+							gender: user.gender,
+							role: user.role,
+							jobRole: user.jobRole,
+							department: user.department,
+							address: user.address,
+							profilePictureUrl: user.profilePictureUrl,
+							createdOn: user.createdOn.toISOString()
```

Comment:
was thinking the database field is rather `createdAt`, i stand corrected

Reply:
so for the transformUserResponse we changed it to createdOn

Reply:
sure but I'm referring to the database column name. the `user` object here is from the database, not sure we changed the field in the database to `createdOn`

Reply:
lemme crosscheck, thanks a lot